src/jit/instrsarm64.h

```diff
@@ -708,6 +708,114 @@ INST1(sturb,   "sturb",  0,ST, IF_LS_2C,  0x38000000)
                                    //  sturh   Rt,[Xn+simm9]        LS_2C  01111000000iiiii iiii00nnnnnttttt   7800 0000   [Xn imm(-256..+255)]
 
+INST1(casb,    "casb",   0, LD|ST, IF_LS_3E,  0x08A07C00)
+                                   //  casb    Wm, Wt, [Xn]         LS_3E  00001000101mmmmm 011111nnnnnttttt   08A0 7C00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(casab,   "casab",  0, LD|ST, IF_LS_3E,  0x08E07C00)
+                                   //  casab   Wm, Wt, [Xn]         LS_3E  00001000111mmmmm 011111nnnnnttttt   08E0 7C00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(casalb,  "casalb", 0, LD|ST, IF_LS_3E,  0x08E0FC00)
+                                   //  casalb  Wm, Wt, [Xn]         LS_3E  00001000111mmmmm 111111nnnnnttttt   08E0 FC00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(caslb,   "caslb",  0, LD|ST, IF_LS_3E,  0x08A0FC00)
+                                   //  caslb   Wm, Wt, [Xn]         LS_3E  00001000101mmmmm 111111nnnnnttttt   08A0 FC00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(cash,    "cash",   0, LD|ST, IF_LS_3E,  0x48A07C00)
+                                   //  cash    Wm, Wt, [Xn]         LS_3E  01001000101mmmmm 011111nnnnnttttt   48A0 7C00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(casah,   "casah",  0, LD|ST, IF_LS_3E,  0x48E07C00)
+                                   //  casah   Wm, Wt, [Xn]         LS_3E  01001000111mmmmm 011111nnnnnttttt   48E0 7C00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(casalh,  "casalh", 0, LD|ST, IF_LS_3E,  0x48E0FC00)
+                                   //  casalh  Wm, Wt, [Xn]         LS_3E  01001000111mmmmm 111111nnnnnttttt   48E0 FC00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(caslh,   "caslh",  0, LD|ST, IF_LS_3E,  0x48A0FC00)
+                                   //  caslh   Wm, Wt, [Xn]         LS_3E  01001000101mmmmm 111111nnnnnttttt   48A0 FC00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(cas,     "cas",    0, LD|ST, IF_LS_3E,  0x88A07C00)
+                                   //  cas     Rm, Rt, [Xn]         LS_3E  1X001000101mmmmm 011111nnnnnttttt   88A0 7C00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(casa,    "casa",   0, LD|ST, IF_LS_3E,  0x88E07C00)
+                                   //  casa    Rm, Rt, [Xn]         LS_3E  1X001000111mmmmm 011111nnnnnttttt   88E0 7C00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(casal,   "casal",  0, LD|ST, IF_LS_3E,  0x88E0FC00)
+                                   //  casal   Rm, Rt, [Xn]         LS_3E  1X001000111mmmmm 111111nnnnnttttt   88E0 FC00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(casl,    "casl",   0, LD|ST, IF_LS_3E,  0x88A0FC00)
+                                   //  casl    Rm, Rt, [Xn]         LS_3E  1X001000101mmmmm 111111nnnnnttttt   88A0 FC00   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(ldaddb,  "ldaddb",  0, LD|ST, IF_LS_3E,  0x38200000)
+                                   //  ldaddb   Wm, Wt, [Xn]        LS_3E  00111000001mmmmm 000000nnnnnttttt   3820 0000   Rm Rt Rn ARMv8.1 LSE Atomics
+
+INST1(ldaddab, "ldaddab", 0, LD|ST, IF_LS_3E,  0x38A00000)
```

Comment:
You could add the store forms for these instruction before or afteer each of the load forms so that the implementations of:
`toLdForm(ins) `and `toStForm(ins) `would be trivial.